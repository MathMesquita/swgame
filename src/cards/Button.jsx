import React from 'react';
import PropTypes from 'prop-types';

const NextButton = ({ clickHandler, label }) => (
  <button className="card-button" onClick={clickHandler}>
    {label}
  </button>
);

NextButton.propTypes = {
  clickHandler: PropTypes.func.isRequired,
  label: PropTypes.string.isRequired,
};

export default NextButton;
