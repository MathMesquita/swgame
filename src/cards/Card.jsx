import React from 'react';
import PropTypes from 'prop-types';

const Card = ({ planet, population, climate, terrain, films }) => (
  <section className="card">
    <h2 className="card-title">{planet}</h2>
    <ul className="card-info">
      <InfoItem title="Population" data={population} />
      <InfoItem title="Climate" data={climate} />
      <InfoItem title="Terrain" data={terrain} />
    </ul>
    <footer className="card-films">featured in {films} films</footer>
  </section>
);

Card.propTypes = {
  planet: PropTypes.string.isRequired,
  population: PropTypes.string.isRequired,
  climate: PropTypes.string.isRequired,
  terrain: PropTypes.string.isRequired,
  films: PropTypes.number.isRequired,
};

export default Card;


const InfoItem = ({ title, data }) => (
  <li className="card-infoItem">
    <span className="card-infoTitle">{title}:</span>
    <span className="card-infoData">{data}</span>
  </li>
);

InfoItem.propTypes = {
  title: PropTypes.string.isRequired,
  data: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
  ]).isRequired,
};
