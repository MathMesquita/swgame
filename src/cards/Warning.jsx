import React from 'react';

import './styles/warning.sass';

const Warning = () => (
  <div className="warning">
    <h2 className="warning-title">WOW !!!</h2>
    <p className="warning-message">You have passed by all the planets, how did you achieved that?
     Well, if you want to restart for another round, click in the button !</p>
  </div>
);

export default Warning;

