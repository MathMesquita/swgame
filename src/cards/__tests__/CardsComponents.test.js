import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import Card from '../Card';
it('renders Card component', () => {
  const card = renderer.create(
    <Card
      planet="Tatooine"
      population="2000000"
      climate="Arid"
      terrain="Desert"
      films={4}
    />
  );
  let tree = card.toJSON();
  expect(tree).toMatchSnapshot();

});

import Button from '../Button';
it('renders Button component', () => {
  const nextBtn = renderer.create(
    <Button
      clickHandler={() => {}}
      label="btntext"
    />,
  );
  let tree = nextBtn.toJSON();

  expect(tree).toMatchSnapshot();
});

import Warning from '../Warning';
it('renders Warning component', () => {
  const warning = renderer.create(
    <Warning />,
  );
  let tree = warning.toJSON();

  expect(tree).toMatchSnapshot();
});