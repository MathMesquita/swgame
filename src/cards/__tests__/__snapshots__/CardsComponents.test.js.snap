exports[`test renders Button component 1`] = `
<button
  className="card-button"
  onClick={[Function]}>
  btntext
</button>
`;

exports[`test renders Card component 1`] = `
<section
  className="card">
  <h2
    className="card-title">
    Tatooine
  </h2>
  <ul
    className="card-info">
    <li
      className="card-infoItem">
      <span
        className="card-infoTitle">
        Population
        :
      </span>
      <span
        className="card-infoData">
        2000000
      </span>
    </li>
    <li
      className="card-infoItem">
      <span
        className="card-infoTitle">
        Climate
        :
      </span>
      <span
        className="card-infoData">
        Arid
      </span>
    </li>
    <li
      className="card-infoItem">
      <span
        className="card-infoTitle">
        Terrain
        :
      </span>
      <span
        className="card-infoData">
        Desert
      </span>
    </li>
  </ul>
  <footer
    className="card-films">
    featured in 
    4
     films
  </footer>
</section>
`;

exports[`test renders Warning component 1`] = `
<div
  className="warning">
  <h2
    className="warning-title">
    WOW !!!
  </h2>
  <p
    className="warning-message">
    You have passed by all the planets, how did you achieved that? Well, if you want to restart for another round, click in the button !
  </p>
</div>
`;
